src/settings.c

```diff
@@ -387,6 +387,102 @@ static void set_logging_settings(json_t *j_section, logging_settings_t *settings
     }
 }
 
+static int set_plugin_settings(json_t *j_plugin_settings,
+                               linked_list_t *plugins_list)
+{
+    linked_list_entry_t *entry;
+    plugin_settings_t *plugin, *plugin_entry;
+    size_t plugin_name_length, plugin_path_length;
+    json_t *j_name, *j_path;
+
+    j_name = json_object_get(j_plugin_settings, "name");
+    j_path = json_object_get(j_plugin_settings, "path");
+
+    if (!json_is_string(j_name))
+    {
+        fprintf(stdout, "Plugin configured without name.\n");
+        return 1;
+    }
+    plugin_name_length = strnlen(json_string_value(j_name),
+                                 J_MAX_LENGTH_PLUGIN_NAME);
+    if (plugin_name_length == 0
+        || plugin_name_length == J_MAX_LENGTH_PLUGIN_NAME)
+    {
+        fprintf(stdout, "Plugin name length is invalid\n");
+        return 1;
+    }
+
+    if (!json_is_string(j_path))
+    {
+        fprintf(stdout, "Plugin configured without path.\n");
+        return 1;
+    }
+    plugin_path_length = strnlen(json_string_value(j_path),
+                                 J_MAX_LENGTH_PLUGIN_NAME);
+    if (plugin_path_length == 0
+        || plugin_path_length == J_MAX_LENGTH_PLUGIN_PATH)
+    {
+        fprintf(stdout, "Plugin path length is invalid\n");
+        return 1;
+    }
+
+    plugin = malloc(sizeof(plugin_settings_t));
+
+    plugin->name = json_string_value(j_name);
+    plugin->path = json_string_value(j_path);
+
+    for (entry = plugins_list->head; entry != NULL; entry = entry->next)
+    {
+        plugin_entry = entry->data;
+        if (strncmp(plugin_entry->name, plugin->name,
+                    J_MAX_LENGTH_PLUGIN_NAME) == 0)
+        {
+            fprintf(stdout,
+                    "Duplicate plugin name \"%s\" entry found.\n",
+                    plugin->name);
+
+            free(plugin);
+            return 1;
+        }
+    }
+
+    linked_list_add(plugins_list, plugin);
+
+    return 0;
+}
+
+static int set_plugins_settings(json_t *j_section,
```

Comment:
`set_plugins_settings()` and `set_plugin_settings()` although makes sense is confusing